docs/api/utility.rst

```diff
@@ -0,0 +1,325 @@
+.. # Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+.. # other BLT Project Developers. See the top-level COPYRIGHT file for details
+.. # 
+.. # SPDX-License-Identifier: (BSD-3-Clause)
+
+Utility Macros
+==============
+
+
+General Utility Macros
+----------------------
+
+
+blt_list_append
+~~~~~~~~~~~~~~~
+
+.. code-block:: cmake
+
+    blt_list_append( TO <list> ELEMENTS [ <element>...] IF <bool> )
+
+Appends elements to a list if the specified bool evaluates to true.
```

Comment:
It might be easier to read if the one-line descriptions were listed above the signatures

Reply:
Trying that in the next build for the first utility.